x/ecocredit/client/tx.go

```diff
@@ -52,14 +57,139 @@ func txCreateClass() *cobra.Command {
 				return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "base64_metadata is malformed, proper base64 string is required")
 			}
 
-			c, ctx, err := mkMsgClient(cmd)
+			c, err := newMsgSrvClient(cmd)
 			if err != nil {
 				return err
 			}
-			_, err = c.CreateClass(cmd.Context(), &ecocredit.MsgCreateClassRequest{
+			msg := ecocredit.MsgCreateClassRequest{
 				Designer: args[0], Issuers: issuers, Metadata: b,
-			})
-			return mkTx(ctx, cmd, err)
+			}
+			_, err = c.client.CreateClass(cmd.Context(), &msg)
+			return c.send(err)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+func txCreateBatch() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "create_batch [issuer] [class_id] [metadata] [issuance]",
+		Short: `Issues new credit batach.
+Parameters:
+  issuer:    issuer address
+  class_id:  credit class
+  metadata:  base64 encoded issuance metadata
+  issuance:  JSON encode issuance list,
+             eg: '[{"recipient": "a1", "tradeableUnits": "10", "retiredUnits": "2"}]'`,
+		Args: cobra.ExactArgs(4),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			b, err := base64.StdEncoding.DecodeString(args[2])
+			if err != nil {
+				return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "base64_metadata is malformed, proper base64 string is required")
+			}
+			var issuance = []*ecocredit.MsgCreateBatchRequest_BatchIssuance{}
+			if err = json.Unmarshal([]byte(args[3]), &issuance); err != nil {
+				return err
+			}
+
+			msg := ecocredit.MsgCreateBatchRequest{
+				Issuer: args[0], ClassId: args[1], Metadata: b, Issuance: issuance,
+			}
+			c, err := newMsgSrvClient(cmd)
+			if err != nil {
+				return err
+			}
+			_, err = c.client.CreateBatch(cmd.Context(), &msg)
+			return c.send(err)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+func txSend() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "send [recipient] [credits]",
+		Short: `Sends given credits to from the transaction author (--from) to the recipient.
+Parameters:
+  recipient: recipient address
+  credits:   JSON encoded credit list
+             eg: '[{"batchDenom": "100/2", "tradeableUnits": "5", "retiredUnits": "0"}]'`,
+		Args: cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var credits = []*ecocredit.MsgSendRequest_SendUnits{}
+			if err := json.Unmarshal([]byte(args[1]), &credits); err != nil {
+				return err
+			}
+			c, err := newMsgSrvClient(cmd)
+			if err != nil {
+				return err
+			}
+			msg := ecocredit.MsgSendRequest{
+				Sender:    c.Cctx.GetFromAddress().String(),
+				Recipient: args[0], Credits: credits,
+			}
+			_, err = c.client.Send(cmd.Context(), &msg)
+			return c.send(err)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+func txRetire() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "retire [holder] [credits]",
+		Short: `Retires specified amount of a holder credits. Transaction author (--from) must
+must be an authorized issuer.
```

Comment:
Do we actually verify that tx author is an issuer? It's not clear to me where this check happens.

Also- for user requirements I think we should also allow for users to retire their own credits (e.g. if --from == holder)

Reply:
Holders retire their own credits that is correct. Issuers shouldn't be mentioned here.